### PR TITLE
[auth] Support IPA/AD Trust

### PR DESCRIPTION
**Pull request by jvlcek**
Documenting:
- Requirements for IPA/AD Trust
- How to configure IPA/AD Trust
- How to use IPA/AD Trust

https://trello.com/c/qq4sTk5o/93-auth-3-support-ipa-ad-trust

### CODE REVIEW

**Comment by jvlcek** [comment]
@abellotti Please review

**Comment by abellotti** [comment]
```
line 8:  how to manually configure an Appliance's   (drop the 7.1)
line 25: 7.1 -> 7.2
line 37: to the following documents:
lines 98, 100:  somehow those show up in red, double check asciidoc.
line 102: add mention to Rest API (and SSUI) ability to authenticate using the AD credentials as well as ability to kinit using AD credentials and ability to SSO to the Admin UI.
```

Other than that looks really good 👍 

Adding @mfeifer who keeps us honest (not too much dev speak 😄 )

**Comment by jvlcek** [comment]
@abellotti Thank you for the feed back. I have address it. I update the usage section to be consistent with the ipa_2fa usage section.

**Comment by jvlcek** [comment]
@mfeifer and @Fryguy Can this be merged please?

File: auth/ipa_ad_trust.adoc [hunk 1]
```diff
@@ -0,0 +1,104 @@
+
+[[ipa-ad-trust]]
+== IPA/AD Trust Authentication
+
+Enabling Active Directory (AD) Trust Authentication on the Appliance is currently supported
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: remove Enabling

File: auth/ipa_ad_trust.adoc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+In this guide we will cover how to manually configure an Appliance's
+external authentication to work with AD Trust Authentication using IPA.
+This provides IPA Users access to the Appliance Administrative UI and the REST API
+using their AD credentials.
+
+1.  <<requirements, Requirements>>
+2.  <<configure-ipa-ad-trust-server, Setup and configure an IPA server for AD Trust Authentication>>
+3.  <<enable-external-auth, Enable the MiQ Appliance to use the configued IPA server>>
+4.  <<configuring-admin-ui, Configuring the MiQ appliance to use external authentication>>
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: change Configuring to Configure

File: auth/ipa_ad_trust.adoc [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+5.  <<create-group-ui, Create groups on the MiQ appliance>>
+6.  <<using-ad-trust, Using AD Trust Authentication>>
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: change Using to Use

File: auth/ipa_ad_trust.adoc [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+[[requirements]]
+== Requirements
+
+The following is needed in order to enable AD Trust Authentication to the Appliance:
+
+* A CentOS/RHEL 7.2 based MiQ Appliance
+* Windows Server 2008 R2 or later with configured AD DC and DNS installed locally on the DC
+
+[[configure-ipa-ad-trust-server]]
+== Setup and configure an IPA server for AD Trust Authentication
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: Set up instead of Setup

File: auth/ipa_ad_trust.adoc [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+* IPA Server based on FreeIPA 3.3.3 or later is recommended
+
+Instructions for setting up and configure cross-realm trust between an IPA domain and an AD (Active Directory) domain
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: configuring instead of configure

File: auth/ipa_ad_trust.adoc [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+is described on http://www.freeipa.org/page/Active_Directory_trust_setup[freeipa.org Active Directory Trust Setup]
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: instead of "is described on" change to "can be found at"

File: auth/ipa_ad_trust.adoc [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+* http://www.freeipa.org/page/Active_Directory_trust_setup#DNS_configuration[freeipa.org Active Directory Trust Setup/DNS configuration]
+* http://www.freeipa.org/page/DNS[freeipa.org DNS]
+* https://www.freeipa.org/page/Howto/DNSSEC[freeipa.org DNSSEC]
+
+[[enable-external-auth]]
+== Enable the MiQ Appliance to use the configued IPA server
+
+Using the Appliance Console, enable external authentication to the IPA server.
+
+1. Login to console as _root_
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: as a noun, it's "login", as a verb it should be two words.

File: auth/ipa_ad_trust.adoc [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+and improper functionality.  For more details refer to the following documents:
+
+* http://www.freeipa.org/page/Active_Directory_trust_setup#DNS_configuration[freeipa.org Active Directory Trust Setup/DNS configuration]
+* http://www.freeipa.org/page/DNS[freeipa.org DNS]
+* https://www.freeipa.org/page/Howto/DNSSEC[freeipa.org DNSSEC]
+
+[[enable-external-auth]]
+== Enable the MiQ Appliance to use the configued IPA server
+
+Using the Appliance Console, enable external authentication to the IPA server.
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: Use the Appliance Console to enable external authentication to the IPA Server.

File: auth/ipa_ad_trust.adoc [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+4. From the Advanced Setting menu, select the menu item *Configure External Authentication (httpd)*
+5. Enter the FQDN of the IPA Server, i.e. _ipaserver.test.company.com_
+6. Enter the IPA Server domain, i.e. _test.company.com_
+7. Enter the IPA Server realm, i.e. _TEST.COMPANY.COM_
+8. Press enter to select the default IPA Server Principal, i.e. _admin_
+9. Enter the Password of the IPA Server Principal
+10. Review details, and Enter *y* to proceed.
+
+[[configuring-admin-ui]]
+== Configuring the MiQ appliance to use external authentication
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: Configure

File: auth/ipa_ad_trust.adoc [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+Login to the MiQ appliance as admin, then in _Settings->Configuration->Server->Authentication_
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: Log in

File: auth/ipa_ad_trust.adoc [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+The above steps need to be done on each UI and WebService enabled appliance.
+
+in _Settings->Configuration->Access Control_
+
+* Make sure the user's groups are created on the Appliance and appropriate roles assigned to those groups.
+
+[[create-group-ui]]
+== Create groups on the MiQ appliance
+
+Login to the MiQ appliance as admin, then in _Settings->Configuration->Access Control->Groups->Configuration->Add a new Group_
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: Log in

File: auth/ipa_ad_trust.adoc [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+* Check: _(Look Up LDAP Groups)_
+* Enter the AD user as the _User to Look Up_ i.e. _ipauser@ipaserver.test.company.com_
+* Click Retrieve.
+* Choose a group from the _LDAP Groups for User_ dropdown.
+* Assign the appropriate roles to the group
+* Click Add.
+
+The above steps need to be done on each UI and WebService enabled appliance.
```
Comment (mfeifer) [inline auth/ipa_ad_trust.adoc]: You should probably move this ABOVE the steps so that they know before they have read it where to perform these steps.

**Comment by jvlcek** [comment]
@mfeifer Thank you for the review feedback. I have address all the concerns you raised.

**Pull request merged**